ci/images/plugin-compiler/Dockerfile

```diff
@@ -17,7 +17,7 @@ COPY ci/images/plugin-compiler/data/build.sh /build.sh
 RUN chmod +x /build.sh
 
 COPY . $TYK_GW_PATH
-RUN cd $TYK_GW_PATH && go mod vendor
```

Comment:
release.yml:L112 has a `go mod vendor` there - has this been removed because it's being done there?

Reply:
This vendored dependencies were moved to GOPATH and then both gateway and plugin  was built with `GO111MODULE=off`. 
But in this PR, we're also trying out an approach with module support enabled, so the vendoring might not be necessary. We try to replace any plugin dependency that overlaps with tyk-gateway dependency to  use the exact version used by tyk-gateway by using `go mod replace` to update the go.mod file and finally replace the gateway dependency with the local gateway source in `$TYK_GATEWAY_PATH` in the plugin compiler image.